lib/puppet/parser/functions/noop.rb

```diff
@@ -1,16 +1,16 @@
-# Set noop to true as default for current and children scopes
-Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default to true for all resources
+# Set noop to true as default for current and children scopes, or 'reset' an inherited default with noop(false)
+Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default for all resources
   in local scope and children scopes. This can be overriden in
   child scopes, or explicitly on each resource.
   ") do |args|
-
+      @noop_value = true if (@noop_value = args[0]).nil? or (![true, false].include? @noop_value)
```

Comment:
Are you referring to passing in a quoted string? 
To maintain backwards compatibility, any arg other than a Boolean false is treated as true. Currently you can pass any arg you like to noop() and it won't complain.
If you would prefer different behavior (e.g. fail on a non-bool/nil arg) then I'm happy to change it.